server/database/queries/users/searchUsers.js

```diff
@@ -1,9 +1,9 @@
 const dbConnection = require('../../config/dbConnection');
 
-exports.searchUsers = () => {
+exports.searchUsers = (query) => {
   const sql = {
-    text: '',
-    values: [],
+    text: 'SELECT users.id, users.username, email, bio, avatar_path, users.native_lang_id, users.learning_lang_id, COALESCE(json_agg(interests) FILTER(WHERE interests.id IS NOT NULL),\'[]\') AS interests FROM users LEFT OUTER JOIN user_interest ON user_interest.user_id = users.id JOIN interests ON interests.id = user_interest.interest_id WHERE users.isactive = true AND users.username LIKE $1 GROUP BY users.id LIMIT 100',
+    values: [`%${query}%`],
```

Comment:
values: [`%${query}%`] 
what it does mean ?

Reply:
and please explain the ...rest in the formatlangV2

Reply:
its called the rest operator
check it here https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/rest_parameters
Rest parameters - JavaScript | MDN

Reply:
the query is hard to explain in text
sql like operator takes something like a regex
and here im inserting the text in the regex that matches the inserted query

'amo' doesnt match 'amoodaa'
'%amo' doesnt match 'amoodaa'
'amo%' matches amoodaa but doesnt match 'mamoodaa'
'%amo%' matches amoodaa and matches 'mamoodaa'